internal/gather/gather_bytes.go

```diff
@@ -166,31 +166,28 @@ func (b *bytesReadSeekCloser) ReadAt(bs []byte, off int64) (int, error) {
 	curSlice := slices[sliceNdx]
 
 	n := copy(bs, curSlice[offset:])
+	m := n
 
 	// if we were able to consume the entire slice then move to the next slice
-	if int64(n) == (int64(len(curSlice)) - offset) {
-		// first chunk written, move on to the next
-		sliceNdx++
-	}
+	// first chunk written, move on to the next
+	sliceNdx++
```

Comment:
@aaron-kasten IIUC, @Shrekster modified these 3 lines because he ran into a bug. Could you folks please double check?

I'm going to re-trigger the providers test in the mean time.